File: cassidoo-newsletter/#311/haskell/Anagram.hs
--Given two strings s and t, return true if t is an anagram of s,
--and false otherwise.

module Anagram where

import Data.Char (toLower)
import Data.List (sort)
import Data.Function (on)

isAnagram :: [Char] -> [Char] -> Bool
isAnagram = (==) `on` (sort . map toLower)

File: cassidoo-newsletter/#311/go/anagram.go
// given two strings `s` and `t` returns `true` if they're anagrams,
// false otherwise
package anagram

import (
	"sort"
	"strings"
)

// Returns the sorted alphabetic symbols of a string in lowercase
func transform(w string) string {
	xs := strings.Split(strings.ToLower(w), "")
	sort.Strings(xs)
	return strings.Join(xs, "")
}

func IsAnagram(s, t string) bool {
	if len(s) != len(t) {
		return false
	}
	return transform(s) == transform(t)
}
